crates/esthri/src/rusoto.rs

```diff
@@ -98,6 +106,52 @@ where
     }
 }
 
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum S3StorageClass {
+    Standard,
+    StandardIA,
+    IntelligentTiering,
+    OneZoneIA,
+    GlacialInstantRetrieval,
+    GlacialFlexibleRetrieval,
+    GlacialDeepArchive,
+    RRS,
+}
+
+impl From<S3StorageClass> for String {
+    fn from(s: S3StorageClass) -> Self {
+        match s {
+            S3StorageClass::Standard => "STANDARD",
+            S3StorageClass::StandardIA => "STANDARD_IA",
+            S3StorageClass::IntelligentTiering => "INTELLIGENT_TIERING",
+            S3StorageClass::OneZoneIA => "ONEZONE_IA",
+            S3StorageClass::GlacialInstantRetrieval => "GLACIER_IR",
+            S3StorageClass::GlacialFlexibleRetrieval => "GLACIER",
+            S3StorageClass::GlacialDeepArchive => "DEEP_ARCHIVE",
+            S3StorageClass::RRS => "REDUCED_REDUNDANCY",
+        }
+        .into()
+    }
+}
```

Comment:
I would implement this conversion via [Display](https://doc.rust-lang.org/std/fmt/trait.Display.html). 

```rust
impl fmt::Display for S3StorageClass {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        match self {
            S3StorageClass::Standard => f.write_str("STANDARD"),
            S3StorageClass::StandardIA => f.write_str("STANDARD_IA"),
            S3StorageClass::IntelligentTiering => f.write_str("INTELLIGENT_TIERING"),
            S3StorageClass::OneZoneIA => f.write_str("ONEZONE_IA"),
            S3StorageClass::GlacialInstantRetrieval => f.write_str("GLACIER_IR"),
            S3StorageClass::GlacialFlexibleRetrieval => f.write_str("GLACIER"),
            S3StorageClass::GlacialDeepArchive => f.write_str("DEEP_ARCHIVE"),
            S3StorageClass::RRS => f.write_str("REDUCED_REDUNDANCY"),
        }
    }
}
```

This provides a few advantages over a pure conversion to strings. Namely you can use this storage class in a format string

```rust
let s = S3StorageClass::RRS;
println!("using {}", s);
```

Instead of 

```rust
let s = S3StorageClass::RRS;
println!("using {}", String::from(s));
```
which will allocate an additional `String` (whereas format arguments are lazy). And if you ever do need an actual `String`, there is a blanket implementation of [ToString](https://doc.rust-lang.org/std/string/trait.ToString.html) for any type that implements `Display`